### DISCUSSION TITLE
`vercel dev` breaks custom imports in vitejs (and possibly other?)

### DISCUSSION THREAD

**Discussion by geirmarius** [vercel/vercel#6145] [Help]
When using vitejs, you can add suffixes to import resources in useful ways (See [vite docs for static asset handling](https://vitejs.dev/guide/assets.html#importing-asset-as-url)).
This is very useful for importing scripts directly as workers, or fetching a resource url that has a different path in dev and prod. However, when you run vite's dev server with vercel's dev server these imports break.

I've been trying to find a solution for a while now, but I just can't seem to understand how this would work together.
Would anyone be able to help me out here? Would be much appreciated! 😊

(Side note: I need to use vercel's dev server on top for access to the serverless functions that are required for the app to run)

**Comment by t3dotgg** [comment]
I have been working on this as well without much luck. It seems as though vite's non-JS imports in dev _request javascript as the response type anyways_

Example: When using an imported SVG in `vite` dev, two fetches occur:
![image](https://user-images.githubusercontent.com/6751787/121792466-e82b5e80-cba9-11eb-977d-cf6ffc6f38df.png)

The first one is a valid JS file `http://localhost:3000/src/assets/TwitchGlitchPurple.svg?import`, which contains the following:
![image](https://user-images.githubusercontent.com/6751787/121792472-f5484d80-cba9-11eb-9258-e3de755e19c6.png)
This JS file has the path to the asset, which is fetched after.

When using `vercel dev`, only one fetch occurs, which directly returns the asset:
![image](https://user-images.githubusercontent.com/6751787/121792486-33457180-cbaa-11eb-8a2c-0cd299315801.png)

It seems like Vite interally uses a `?import` query param to override asset paths with a JS shim as seen above. 

Console error of import type mismatch:
![image](https://user-images.githubusercontent.com/6751787/121792504-8cada080-cbaa-11eb-98c4-c19f77e20572.png)

Not sure what the best path forward here is. My _best guess_ is that `vercel dev` tries to serve static assets itself, ignoring query params. Perhaps query params should break the match and try to pass the request on to the frontend framework? 🤔
